src/Microsoft.IdentityModel.Logging/TextWriterEventListener.cs

```diff
@@ -0,0 +1,71 @@
+﻿//-----------------------------------------------------------------------
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved
+// Apache License 2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------
+
+using System.Diagnostics.Tracing;
+using System.IO;
+
+namespace Microsoft.IdentityModel.Logging
+{
+    /// <summary>
+    /// Event listener that writes logs to a file or a fileStream provided by user.
+    /// </summary>
+    public class TextWriterEventListener : EventListener
+    {
+        private static string _defaultLogFileName = "IdentityModelLogs.txt";
+        private StreamWriter _streamWriter;
+        private Stream _fileStream;
+        private string _filePath;
+
+        public TextWriterEventListener()
+        {
+            _fileStream = new FileStream(_defaultLogFileName, FileMode.OpenOrCreate, FileAccess.Write);
+            _streamWriter = new StreamWriter(_fileStream);
+        }
+
+        public TextWriterEventListener(string filePath)
+        {
+            _filePath = filePath;
+            _fileStream = new FileStream(_filePath, FileMode.OpenOrCreate, FileAccess.Write);
+            _streamWriter = new StreamWriter(_fileStream);
+        }
+
+        public TextWriterEventListener(Stream fileStream)
+        {
+            _fileStream = fileStream;
+            _streamWriter = new StreamWriter(_fileStream);
+        }
+
+        protected override void OnEventWritten(EventWrittenEventArgs eventData)
+        {
+            if (eventData.Payload == null || eventData.Payload.Count <= 0)
+                return;
+
+            for (int i = 0; i < eventData.Payload.Count; i++)
+            {
+                _streamWriter.WriteLine(eventData.Payload[i].ToString());
+            }
+        }
+
+        public override void Dispose()
+        {
+            _streamWriter.Dispose();
+            _fileStream.Dispose();
```

Comment:
you can't call .Dispose on the _streamWriter or _fileStream as you may not own them.
However, I don't think we should have the Constructor that takes a FileStream.